libs/SalesforceHybridSDK/SalesforceHybridSDK/Classes/Plugins/SFSmartSyncPlugin/SFSmartSyncPlugin.m

```diff
@@ -87,27 +87,31 @@ - (void)handleSyncUpdate:(SFSyncState*)sync isGlobal:(BOOL)isGlobal
 {
     dispatch_async(dispatch_get_main_queue(), ^{
         NSError *error = nil;
-        NSMutableDictionary* detailDict = [NSMutableDictionary dictionaryWithDictionary:[sync asDict]];
-        detailDict[kSyncIsGlobalStoreArg] = isGlobal ? @YES : @NO;
-        NSData *detailData = [NSJSONSerialization dataWithJSONObject:detailDict
-                                                             options:0 // non-pretty printing
-                                                               error:&error];
-        if(error) {
-            [self log:SFLogLevelError format:@"JSON Parsing Error: %@", error];
-        }
-        else {
-            NSString* detailAsString = [[NSString alloc] initWithData:detailData encoding:NSUTF8StringEncoding];
-            NSString* js = [
-                            @[@"document.dispatchEvent(new CustomEvent(\"",
-                              kSyncEventType,
-                              @"\", { \"",
-                              kSyncDetail,
-                              @"\": ",
-                              detailAsString,
-                              @"}))" ]
-                            componentsJoinedByString:@""
-                            ];
-            [self.commandDelegate evalJs:js];
+        if ([NSJSONSerialization isValidJSONObject:[sync asDict]]) {
```

Comment:
Doc suggests checking validity of JSON before attempting to parse. This was fixed in other places a while ago, but this one was left out.